qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -6,646 +6,14 @@
-    local image=$3
-    local snap=$4
-
-    rbd --cluster ${cluster} -p ${pool} snap unprotect ${image}@${snap}
-}
-
-wait_for_snap_present()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local snap_name=$4
-    local s
-
-    for s in 1 2 4 8 8 8 8 8 8 8 8 16 16 16 16 32 32 32 32; do
-	sleep ${s}
-        rbd --cluster ${cluster} -p ${pool} info ${image}@${snap_name} || continue
-        return 0
-    done
-    return 1
-}
-
-write_image()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local count=$4
-
-    rbd --cluster ${cluster} -p ${pool} bench-write ${image} \
-	--io-size 4096 --io-threads 1 --io-total $((4096 * count)) \
-	--io-pattern rand
-}
-
-compare_images()
-{
-    local pool=$1
-    local image=$2
-
-    local rmt_export=${TEMPDIR}/${CLUSTER2}-${pool}-${image}.export
-    local loc_export=${TEMPDIR}/${CLUSTER1}-${pool}-${image}.export
-
-    rm -f ${rmt_export} ${loc_export}
-    rbd --cluster ${CLUSTER2} -p ${pool} export ${image} ${rmt_export}
-    rbd --cluster ${CLUSTER1} -p ${pool} export ${image} ${loc_export}
-    cmp ${rmt_export} ${loc_export}
-}
-
-demote_image()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image demote ${pool}/${image}
-}
-
-promote_image()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image promote ${pool}/${image}
-}
-
-set_pool_mirror_mode()
-{
-    local cluster=$1
-    local pool=$2
-    local mode=$3
-
-    rbd --cluster=${cluster} -p ${pool} mirror pool enable ${mode}
-}
-
-disable_mirror()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image disable ${pool}/${image}
-}
-
-enable_mirror()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image enable ${pool}/${image}
-}
-
-test_image_present()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local test_state=$4
-    local current_state=deleted
-
-    rbd --cluster=${cluster} -p ${pool} ls | grep "^${image}$" &&
-    current_state=present
-
-    test "${test_state}" = "${current_state}"
-}
-
-wait_for_image_present()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local state=$4
-    local s
-
-    # TODO: add a way to force rbd-mirror to update replayers
-    for s in 1 2 4 8 8 8 8 8 8 8 8 16 16; do
-	sleep ${s}
-	test_image_present "${cluster}" "${pool}" "${image}" "${state}" && return 0
-    done
-    return 1
-}
-
-#
-# Main
-#
-
-if [ "$#" -gt 0 ]
-then
-    if [ -z "${RBD_MIRROR_TEMDIR}" ]
-    then
-       echo "RBD_MIRROR_TEMDIR is not set" >&2
-       exit 1
-    fi
-
-    TEMPDIR="${RBD_MIRROR_TEMDIR}"
-    cd ${TEMPDIR}
-    $@
-    exit $?
+if [ -n "${CEPH_REF}" ]; then
+  wget -O rbd_mirror_helpers.sh "https://git.ceph.com/?p=ceph.git;a=blob_plain;hb=$CEPH_REF;f=qa/workunits/rbd/rbd_mirror_helpers.sh"
+  . rbd_mirror_helpers.sh
```

Comment:
@trociny Thanks for that -- I saw that it caused failures in my jewel testing branch.  I am rebuilding w/o the improved tests for now so that I can re-run the failures.